### PR TITLE
files: add file collections editor in the admin view 

### PR DESCRIPTION
**Pull request by jma**

### CODE REVIEW

File: projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+@if (record) {
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.html:1]: Missing copyright

File: projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.scss [hunk 2]
```diff
@@ -0,0 +1,6 @@
+:host ::ng-deep {
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.scss:1]: Missing copyright

File: projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.spec.ts:1]: Missing copyright

File: projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { Component, Input, OnDestroy, OnInit, inject } from '@angular/core';
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts:1]: Missing copyright

File: projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  private resourcesFilesService = inject(FilesService);
+  private toastrService = inject(ToastrService);
+  private translateService = inject(TranslateService);
+
+
+  @Input() type;
+  @Input() pid;
+  record: any;
+  collections = new FormControl('');
+  private subscriptions = new Subscription();
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts:24]: docstring

File: projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+
+  ngOnInit(): void {
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts:27]: docstring

File: projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    this.subscriptions.add(this.resourcesFilesService.currentParentRecord$.subscribe(record => {
+      this.setRecord(record);
+    }));
+  }
+
+  ngOnDestroy(): void {
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts:33]: docstring

File: projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+   this.subscriptions.unsubscribe();
+  }
+  setRecord(record) {
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts:36]: missing return and docstring

File: projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    this.record = record;
+    if (this.record?.metadata?.collections) {
+      this.collections.setValue(this.record.metadata.collections.join(', '));
+    } else {
+      this.collections.setValue('');
+    }
+  }
+  disabled() {
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts:44]: missing return and docstring

File: projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+    this.collections.touched
+  }
+  save() {
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts:47]: missing return and doctring

File: projects/admin/src/app/service/resources-files.service.spec.ts [hunk 11]
```diff
@@ -0,0 +1,21 @@
+import { TestBed } from '@angular/core/testing';
```
Comment (Garfield-fr) [inline projects/admin/src/app/service/resources-files.service.spec.ts:1]: Missing copyright

File: projects/admin/src/app/service/resources-files.service.ts [hunk 12]
```diff
@@ -0,0 +1,311 @@
+import { HttpClient, HttpResponse } from "@angular/common/http";
```
Comment (Garfield-fr) [inline projects/admin/src/app/service/resources-files.service.ts:1]: Missing copyright

File: projects/admin/src/app/service/resources-files.service.ts [hunk 13]
```diff
@@ -0,0 +1,311 @@
...
+          links: [`doc_${pid}`],
+          owners: [`lib_${libPid}`],
+        },
+      })
+      .pipe(
+        tap((record) => this.currentParentRecord.next(record))
+      ) as Observable<Record>;
+  }
+
+  updateParentRecordMetadata(
```
Comment (Garfield-fr) [inline projects/admin/src/app/service/resources-files.service.ts:95]: docstring

File: projects/shared/src/lib/component/documents/files/files.component.html [hunk 14]
```diff
@@ -0,0 +1,53 @@
+<!--
+  RERO ILS UI
+  Copyright (C) 2021 RERO
```
Comment (Garfield-fr) [inline projects/shared/src/lib/component/documents/files/files.component.html:3]: 2024 ?

File: projects/shared/src/lib/component/documents/files/files.component.ts [hunk 15]
```diff
@@ -0,0 +1,149 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2019-2023 RERO
```
Comment (Garfield-fr) [inline projects/shared/src/lib/component/documents/files/files.component.ts:3]: 2024

File: projects/shared/src/lib/component/documents/files/files.component.ts [hunk 16]
```diff
@@ -0,0 +1,149 @@
...
+          });
+          Object.values(data).map((d: File) => files.push(d));
+        });
+        this.files = files.sort((a, b) => a.label.localeCompare(b.label));
+      });
+  }
+
+  /** invenio previewer
+   *
+   * TODO
```
Comment (Garfield-fr) [inline projects/shared/src/lib/component/documents/files/files.component.ts:146]: ?

File: projects/shared/src/lib/modules/prime-ng-import/prime-ng-import.module.ts [hunk 17]
```diff
@@ -1,6 +1,6 @@
 /*
  * RERO ILS UI
- * Copyright (C) 2019-2024 RERO
+ * Copyright (C) 2019-2022 RERO
```
Comment (Garfield-fr) [inline projects/shared/src/lib/modules/prime-ng-import/prime-ng-import.module.ts:3]: 2024

**Review by Garfield-fr — APPROVED** [review]

File: projects/shared/src/lib/component/documents/files/files.component.ts (hunk 16 above)
Comment (jma) [reply projects/shared/src/lib/component/documents/files/files.component.ts:146]: Will be implemented in a next PR.

**Review by lauren-d — APPROVED** [review]

**Pull request merged**